hikari/impl/buckets.py

```diff
@@ -267,8 +265,34 @@ def acquire(self) -> asyncio.Future[None]:
         asyncio.Future[builtins.None]
             A future that should be awaited immediately. Once the future completes,
             you are allowed to proceed with your operation.
+
+        Raises
+        ------
+        builtins.RuntimeError
+            If the lock is not acquired.
+        hikari.errors.RateLimitTooLongError
+            If the ratelimit is longer than `max_rate_limit`.
         """
-        return aio.completed_future(None) if self.is_unknown else super().acquire()
+        if not self._lock.locked():
+            raise RuntimeError("Lock is not acquired")
```

Comment:
Multiple things.
1: just because the lock is locked doesn't mean this is the place where it's acquired?
2:  Doesn't this break the interface rules of the base ratelimiter since it makes the acquire method almost guaranteed to fail unless something else is also acquiring the ratelimiter with async with syntax at the same time.
3: shouldn't the acquire method be the thing which acquires the lock? isn't the whole async with syntax meant to be a short hand for what could otherwise be achieved through method calls rather than something which needs to directly used to achieve something.

Reply:
1. I don't think there is a way to check that they are both in the same place. The idea behind this is to not allow acquiring the bucket without acquiring the lock before.

2. Yea, kinda. What I disliked about your approach to this was having to change the whole system to match the interface of this specific object.

3. Acquire is a blocking method, meaning that it wont be able to acquire the asyncio lock in any way. Furthermore, using an async context manager allows us to keep the bucket locked until we are done updating it or we error out, in which cause it will just be released.

Reply:
Changed in the next commit